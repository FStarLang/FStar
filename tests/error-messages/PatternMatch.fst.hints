[
  "!\bZ\nñKDá\u001d\t\u0002:àòZê",
  [
    [
      "PatternMatch.uu___14",
      1,
      2,
      1,
      [ "@query", "equation_PatternMatch.uu___13" ],
      0,
      "55e56eada0b61200811112861f0f4ff1"
    ],
    [
      "PatternMatch.uu___45",
      1,
      2,
      1,
      [
        "@MaxIFuel_assumption", "@query",
        "Prims_pretyping_ae567c2fb75be05905677af440075565",
        "equation_PatternMatch.uu___31", "equation_PatternMatch.x1",
        "function_token_typing_Prims.__cache_version_number__",
        "projection_inverse_FStar.Pervasives.Native.Mktuple2__1"
      ],
      0,
      "753d9ea3511f01a496224cf734784c07"
    ],
    [
      "PatternMatch.uu___46",
      1,
      2,
      1,
      [
        "@MaxIFuel_assumption", "@query",
        "Prims_pretyping_ae567c2fb75be05905677af440075565",
        "equation_PatternMatch.uu___31", "equation_PatternMatch.y1",
        "function_token_typing_Prims.__cache_version_number__",
        "projection_inverse_FStar.Pervasives.Native.Mktuple2__2"
      ],
      0,
      "27cb13b98dd8bc0044c681d2cde631a1"
    ],
    [
      "PatternMatch.uu___47",
      1,
      2,
      1,
      [
        "@MaxIFuel_assumption", "@query",
        "Prims_pretyping_ae567c2fb75be05905677af440075565",
        "equation_PatternMatch.uu___38", "equation_PatternMatch.x2",
        "function_token_typing_Prims.__cache_version_number__",
        "projection_inverse_FStar.Pervasives.Native.Mktuple2__1"
      ],
      0,
      "b918a8e27d24012aea362d3b6c74557d"
    ],
    [
      "PatternMatch.uu___48",
      1,
      2,
      1,
      [
        "@MaxIFuel_assumption", "@query",
        "Prims_pretyping_ae567c2fb75be05905677af440075565",
        "equation_PatternMatch.uu___38", "equation_PatternMatch.y2",
        "function_token_typing_Prims.__cache_version_number__",
        "projection_inverse_FStar.Pervasives.Native.Mktuple2__2"
      ],
      0,
      "1749194e07c103182813c37230a13cca"
    ],
    [
      "PatternMatch.x",
      1,
      2,
      1,
      [
        "@query", "equation_PatternMatch.uu___69",
        "proj_equation_FStar.Pervasives.Native.Mktuple2__2",
        "projection_inverse_FStar.Pervasives.Native.Mktuple2__2"
      ],
      0,
      "c2f6cdfa85f7d945a2f42b4d34e32cde"
    ],
    [
      "PatternMatch.z",
      1,
      2,
      1,
      [
        "@query", "equation_PatternMatch.uu___78",
        "proj_equation_FStar.Pervasives.Native.Mktuple2__2",
        "projection_inverse_FStar.Pervasives.Native.Mktuple2__2"
      ],
      0,
      "d53324ce0808962e30d40f9d95520fa1"
    ]
  ]
]