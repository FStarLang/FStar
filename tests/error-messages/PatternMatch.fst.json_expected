>> Got issues: [
{"msg":["Type ascriptions within patterns are only allowed on variables"],"level":"Error","range":{"def":{"file_name":"PatternMatch.fst","start_pos":{"line":15,"col":27},"end_pos":{"line":15,"col":34}},"use":{"file_name":"PatternMatch.fst","start_pos":{"line":15,"col":27},"end_pos":{"line":15,"col":34}}},"number":178,"ctx":["While desugaring module PatternMatch"]}
>>]
>> Got issues: [
{"msg":["Type ascriptions within patterns are only allowed on variables"],"level":"Error","range":{"def":{"file_name":"PatternMatch.fst","start_pos":{"line":18,"col":29},"end_pos":{"line":18,"col":48}},"use":{"file_name":"PatternMatch.fst","start_pos":{"line":18,"col":29},"end_pos":{"line":18,"col":48}}},"number":178,"ctx":["While desugaring module PatternMatch"]}
>>]
>> Got issues: [
{"msg":["Type ascriptions within patterns are only allowed on variables"],"level":"Error","range":{"def":{"file_name":"PatternMatch.fst","start_pos":{"line":41,"col":3},"end_pos":{"line":41,"col":24}},"use":{"file_name":"PatternMatch.fst","start_pos":{"line":41,"col":3},"end_pos":{"line":41,"col":24}}},"number":178,"ctx":["While desugaring module PatternMatch"]}
>>]
>> Got issues: [
{"msg":["Patterns are incomplete","The SMT solver could not prove the query. Use --query_stats for more details.","Also see: PatternMatch.fst(21,4-21,8)",""],"level":"Error","range":{"def":{"file_name":"dummy","start_pos":{"line":0,"col":0},"end_pos":{"line":0,"col":0}},"use":{"file_name":"dummy","start_pos":{"line":0,"col":0},"end_pos":{"line":0,"col":0}}},"number":19,"ctx":["While checking for top-level effects","While typechecking the top-level declaration `let uu___7`","While typechecking the top-level declaration `[@@expect_failure] let uu___6`"]}
>>]
>> Got issues: [
{"msg":["Patterns are incomplete","The SMT solver could not prove the query. Use --query_stats for more details.","Also see: PatternMatch.fst(24,4-24,11)",""],"level":"Error","range":{"def":{"file_name":"dummy","start_pos":{"line":0,"col":0},"end_pos":{"line":0,"col":0}},"use":{"file_name":"dummy","start_pos":{"line":0,"col":0},"end_pos":{"line":0,"col":0}}},"number":19,"ctx":["While checking for top-level effects","While typechecking the top-level declaration `let uu___10`","While typechecking the top-level declaration `[@@expect_failure] let uu___9`"]}
>>]
>> Got issues: [
{"msg":["Patterns are incomplete","The SMT solver could not prove the query. Use --query_stats for more details.","Also see: PatternMatch.fst(29,4-29,8)",""],"level":"Error","range":{"def":{"file_name":"dummy","start_pos":{"line":0,"col":0},"end_pos":{"line":0,"col":0}},"use":{"file_name":"dummy","start_pos":{"line":0,"col":0},"end_pos":{"line":0,"col":0}}},"number":19,"ctx":["While checking for top-level effects","While typechecking the top-level declaration `let uu___17`","While typechecking the top-level declaration `[@@expect_failure] let uu___16`"]}
>>]
>> Got issues: [
{"msg":["Type of pattern (Prims.bool) does not match type of scrutinee (Prims.int)"],"level":"Error","range":{"def":{"file_name":"PatternMatch.fst","start_pos":{"line":32,"col":4},"end_pos":{"line":32,"col":9}},"use":{"file_name":"PatternMatch.fst","start_pos":{"line":32,"col":4},"end_pos":{"line":32,"col":9}}},"number":114,"ctx":["While typechecking the top-level declaration `let uu___20`","While typechecking the top-level declaration `[@@expect_failure] let uu___19`"]}
>>]
>> Got issues: [
{"msg":["Type of pattern PatternMatch.ab\ndoes not match type of scrutinee Prims.int","Head mismatch PatternMatch.ab vs Prims.int"],"level":"Error","range":{"def":{"file_name":"PatternMatch.fst","start_pos":{"line":35,"col":4},"end_pos":{"line":35,"col":5}},"use":{"file_name":"PatternMatch.fst","start_pos":{"line":35,"col":4},"end_pos":{"line":35,"col":5}}},"number":114,"ctx":["While typechecking the top-level declaration `let uu___23`","While typechecking the top-level declaration `[@@expect_failure] let uu___22`"]}
>>]
>> Got issues: [
{"msg":["Patterns are incomplete","The SMT solver could not prove the query. Use --query_stats for more details.","Also see: PatternMatch.fst(38,4-38,5)",""],"level":"Error","range":{"def":{"file_name":"dummy","start_pos":{"line":0,"col":0},"end_pos":{"line":0,"col":0}},"use":{"file_name":"dummy","start_pos":{"line":0,"col":0},"end_pos":{"line":0,"col":0}}},"number":19,"ctx":["While checking for top-level effects","While typechecking the top-level declaration `let uu___26`","While typechecking the top-level declaration `[@@expect_failure] let uu___25`"]}
>>]
>> Got issues: [
{"msg":["Type of pattern (Prims.bool) does not match type of scrutinee (Prims.int)"],"level":"Error","range":{"def":{"file_name":"PatternMatch.fst","start_pos":{"line":55,"col":4},"end_pos":{"line":55,"col":8}},"use":{"file_name":"PatternMatch.fst","start_pos":{"line":55,"col":4},"end_pos":{"line":55,"col":8}}},"number":114,"ctx":["While typechecking the top-level declaration `let uu___54`","While typechecking the top-level declaration `[@@expect_failure] let uu___53`"]}
>>]
>> Got issues: [
{"msg":["Type of pattern (Prims.bool) does not match type of scrutinee (Prims.int)"],"level":"Error","range":{"def":{"file_name":"PatternMatch.fst","start_pos":{"line":58,"col":4},"end_pos":{"line":58,"col":16}},"use":{"file_name":"PatternMatch.fst","start_pos":{"line":58,"col":4},"end_pos":{"line":58,"col":16}}},"number":114,"ctx":["While typechecking the top-level declaration `let uu___57`","While typechecking the top-level declaration `[@@expect_failure] let uu___56`"]}
>>]
>> Got issues: [
{"msg":["Patterns are incomplete","The SMT solver could not prove the query. Use --query_stats for more details.","Also see: PatternMatch.fst(61,5-61,12)",""],"level":"Error","range":{"def":{"file_name":"dummy","start_pos":{"line":0,"col":0},"end_pos":{"line":0,"col":0}},"use":{"file_name":"dummy","start_pos":{"line":0,"col":0},"end_pos":{"line":0,"col":0}}},"number":19,"ctx":["While checking for top-level effects","While typechecking the top-level declaration `let x`","While typechecking the top-level declaration `[@@expect_failure] let uu___60`"]}
>>]
>> Got issues: [
{"msg":["Patterns are incomplete","The SMT solver could not prove the query. Use --query_stats for more details.","Also see: PatternMatch.fst(64,5-64,20)",""],"level":"Error","range":{"def":{"file_name":"dummy","start_pos":{"line":0,"col":0},"end_pos":{"line":0,"col":0}},"use":{"file_name":"dummy","start_pos":{"line":0,"col":0},"end_pos":{"line":0,"col":0}}},"number":19,"ctx":["While checking for top-level effects","While typechecking the top-level declaration `let x`","While typechecking the top-level declaration `[@@expect_failure] let uu___64`"]}
>>]
>> Got issues: [
{"msg":["Type of pattern (Prims.bool) does not match type of scrutinee (Prims.int)"],"level":"Error","range":{"def":{"file_name":"PatternMatch.fst","start_pos":{"line":69,"col":5},"end_pos":{"line":69,"col":15}},"use":{"file_name":"PatternMatch.fst","start_pos":{"line":69,"col":5},"end_pos":{"line":69,"col":15}}},"number":114,"ctx":["While typechecking the top-level declaration `let uu___74`","While typechecking the top-level declaration `[@@expect_failure] let uu___73`"]}
>>]
Verified module: PatternMatch
All verification conditions discharged successfully
