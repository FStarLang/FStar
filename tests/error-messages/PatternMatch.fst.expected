>> Got issues: [
(Error 19) Patterns are incomplete (Also see: PatternMatch.fst(21,4-21,8)); The SMT solver could not prove the query. Use --query_stats for more details.
>>]
>> Got issues: [
(Error 19) Patterns are incomplete (Also see: PatternMatch.fst(24,4-24,11)); The SMT solver could not prove the query. Use --query_stats for more details.
>>]
>> Got issues: [
(Error 19) Patterns are incomplete (Also see: PatternMatch.fst(29,4-29,8)); The SMT solver could not prove the query. Use --query_stats for more details.
>>]
>> Got issues: [
PatternMatch.fst(32,4-32,9): (Error 114) Type of pattern (Prims.bool) does not match type of scrutinee (Prims.int)
>>]
>> Got issues: [
PatternMatch.fst(35,4-35,5): (Error 114) Type of pattern (PatternMatch.ab) does not match type of scrutinee (Prims.int); head mismatch PatternMatch.ab vs Prims.int
>>]
>> Got issues: [
(Error 19) Patterns are incomplete (Also see: PatternMatch.fst(38,4-38,5)); The SMT solver could not prove the query. Use --query_stats for more details.
>>]
>> Got issues: [
PatternMatch.fst(55,4-55,8): (Error 114) Type of pattern (Prims.bool) does not match type of scrutinee (Prims.int)
>>]
>> Got issues: [
PatternMatch.fst(58,4-58,16): (Error 114) Type of pattern (Prims.bool) does not match type of scrutinee (Prims.int)
>>]
>> Got issues: [
(Error 19) Patterns are incomplete (Also see: PatternMatch.fst(61,5-61,12)); The SMT solver could not prove the query. Use --query_stats for more details.
>>]
>> Got issues: [
(Error 19) Patterns are incomplete (Also see: PatternMatch.fst(64,5-64,20)); The SMT solver could not prove the query. Use --query_stats for more details.
>>]
>> Got issues: [
PatternMatch.fst(69,5-69,15): (Error 114) Type of pattern (Prims.bool) does not match type of scrutinee (Prims.int)
>>]
Verified module: PatternMatch
All verification conditions discharged successfully
